### PR TITLE
7: Submit timesheet alert.

### PR DESCRIPTION
**Pull request by robherba**
**Steps to test**
- Create some nodes: users, project, task, client, timesheet and time entries.
- Check if the button 'Submit Timesheet' run successfully.
- Check if the timesheet's submit form show an alert when you register less hours than the weekly hours.

[issue](https://github.com/ManatiCR/manatime/issues/7)

### CODE REVIEW

File: modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module [hunk 1]
```diff
@@ -126,15 +134,8 @@ function manatime_feature_timesheets_cron() {
...
-      $timesheet->title = 'Timesheet from: ' . $beginning->format('F d, Y') . ' to: ' . gmdate('F d, Y', $end);
-      $timesheet->type = 'timesheet';
-      $timesheet->field_date[LANGUAGE_NONE][0]['value'] = $beginning->format('Y-m-d H:i:s');
-      $timesheet->field_date[LANGUAGE_NONE][0]['value2'] = gmdate('Y-m-d H:i:s', $end);
-      $timesheet->field_user[LANGUAGE_NONE][0]['target_id'] = $uid;
-      $timesheet->status = 0;
-      node_object_prepare($timesheet);
-      node_save($timesheet);
+      $title = 'Timesheet from: ' . $beginning->format('F d, Y') . ' to: ' . gmdate('F d, Y', $end);
+      create_timesheet($title, $beginning->format('Y-m-d H:i:s'), gmdate('Y-m-d H:i:s', $end), $uid);
```
Comment (kporras07) [inline modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module]: Rename function

File: modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module [hunk 2]
```diff
@@ -155,7 +156,7 @@ function manatime_feature_timesheets_cron() {
 }
 
 /**
- * Send an email to each of the users who have not submitted the timesheet.
+ * Send an email to each user who have not submitted the timesheet.
```
Comment (kporras07) [inline modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module]: who has not submitted the timesheet.

File: modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module [hunk 3]
```diff
@@ -279,12 +293,47 @@ function manatime_feature_timesheets_remove_row_form($form, &$form_state) {
...
+}
+
+/**
  * Helper function to make a link add row.
  */
 function manatime_feature_timesheet_make_link_add_row($node) {
   ctools_include('modal');
   ctools_modal_add_js();
-  return l(t('Add row'), 'timesheet/add-new-form/' . $node->nid . '/nojs', array('attributes' => array('class' => 'ctools-use-modal')));
+  if ($node->status == NODE_NOT_PUBLISHED) {
```
Comment (kporras07) [inline modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module:334]: `===`

File: modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module [hunk 4]
```diff
@@ -293,7 +342,20 @@ function manatime_feature_timesheet_make_link_add_row($node) {
 function manatime_feature_timesheet_make_link_remove_row($node, $id_task) {
   ctools_include('modal');
   ctools_modal_add_js();
-  return l(t('Remove row'), 'timesheet/remove-row-form/' . $node->nid . '/' . $id_task . '/nojs', array('attributes' => array('class' => 'ctools-use-modal')));
+  if ($node->status == NODE_NOT_PUBLISHED) {
```
Comment (kporras07) [inline modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module:345]: Same here

File: modules/features/manatime_feature_timesheets/plugins/relationships/timesheet_node.inc [hunk 5]
```diff
@@ -14,33 +14,82 @@ $plugin = array(
...
-  if (date('w') == 1) {
-    $beginning = new DateTime('now');
+
+  $uid = $context[0]->data->uid;
+  // If the url does not have dates, get the current week's timesheet.
+  if (empty($context[1]->data)) {
+    $beginning = date('w') == 1 ? new DateTime('now') : new DateTime('last monday');
+    /*
+     * $end = $beginning->getTimestamp()(Monday's number of seconds)
+     * + 576000(seven days in seconds).
```
Comment (kporras07) [inline modules/features/manatime_feature_timesheets/plugins/relationships/timesheet_node.inc]: Just explain what is 576000.

File: modules/features/manatime_feature_timesheets/plugins/relationships/timesheet_node.inc [hunk 6]
```diff
@@ -14,33 +14,82 @@ $plugin = array(
...
+ */
+function timesheet_node_from_user_settings_form($form, $form_state) {
+  $conf = $form_state['conf'];
+  return $form;
+}
+
+/**
+ * Get timesheet.
+ *
+ * @param string $status
```
Comment (kporras07) [inline modules/features/manatime_feature_timesheets/plugins/relationships/timesheet_node.inc]: Not string.

File: modules/features/manatime_feature_timesheets/plugins/relationships/timesheet_node.inc [hunk 7]
```diff
@@ -14,33 +14,82 @@ $plugin = array(
...
+ * @param string $beginning
+ *   Beginning date, possible values: date or array(dates).
+ * @param string $end
+ *   End date, possible values: date or array(dates).
+ * @param string $condition
+ *   Condition to compare, possible values: '=' or 'BETWEEN'.
+ * @param string $uid
+ *   The timesheet user's id owner.
+ *
+ * @return node
```
Comment (kporras07) [inline modules/features/manatime_feature_timesheets/plugins/relationships/timesheet_node.inc]: object

File: modules/features/manatime_feature_timesheets/themes/timesheet-view.tpl.php [hunk 8]
```diff
@@ -7,6 +7,7 @@
  ?>
 
  <main id="time-entries-main">
+
```
Comment (kporras07) [inline modules/features/manatime_feature_timesheets/themes/timesheet-view.tpl.php]: remove

**Review by kporras07 — CHANGES_REQUESTED** [review]

File: modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module (hunk 3 above)
Comment (robherba) [reply modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module:334]: The state is a string, '===' does not work.

File: modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module (hunk 4 above)
Comment (robherba) [reply modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module:345]: The state is a string, '===' does not work.

File: modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module [hunk 9]
```diff
@@ -600,7 +600,7 @@ function manatime_feature_timesheets_submit_form_submit(&$form, &$form_state) {
 /**
  * Create a timesheet node.
  */
-function create_timesheet($title, $beginning, $end, $uid) {
+function manatime_create_timesheet($title, $beginning, $end, $uid) {
```
Comment (kporras07) [inline modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module]: prefix function with module name please. `manatime_feature_timesheet_create_timesheet`

**Review by kporras07 — CHANGES_REQUESTED** [review]
Really really minor change.

**Comment by robherba** [comment]
All done.

**Review by kporras07 — APPROVED** [review]

**Pull request merged**